single_tech_samples/storage/sample1_data_lifecycle/README.md

```diff
@@ -0,0 +1,224 @@
+
+Users with appropriate access rights should be able to:
+
+- View the storage accounts in the Azure portal.
+- View the storage containers under the storage accounts through the Azure portal.
+- View the data lifecycle rules created in the Azure portal under storage accounts.
+
+Here are the detail steps on how to execute the sample:
+
+- Checkout the code.
+- Configure the variables as needed under the files variables.tf and terraform.auto.tfvars
+- Here are the sequence of commands to run from the main directory :
+
+   ```bash
+   terraform init
+   terraform plan
+   terraform apply
+   ```
+
+After you run the above commands successfully, you should be able to see the following items under your Azure Subscription:
+
+- Azure Storage Accounts. Sample view is shown below.
+
+![alt text](../common_assets/images/storage_accounts.png "Logo Title Text 1")
+You can observe the appended random strings to make the storage accounts’ names unique since we have used the azurecaf feature.
+
+- Azure Storage Containers. Sample view is shown below.
+
+![alt text](../common_assets/images/storage_containers.png "Logo Title Text 1")
+
+- Azure data lifecycle rules. Sample view is shown below.
+
+![alt text](../common_assets/images/data_lifecycle_rules.png "Logo Title Text 1")
+
+### 2.5. Clean-up
+
+Please follow the below steps to clean up your environment :
+
+The clean-up script can be executed to clean up the resources provisioned in this sample. Following are the steps to execute the script:
+
+- Navigate to `single_tech_samples/synapseanalytics/sample1_loading_dynamic_modules/setup`.
+
+- Run terraform destroy command :
+
+   ```bash
+   terraform destroy
+   ```
+
+## 3. Troubleshooting
+
+### Error Building AzureRM Client
+
+If you run the terraform plan command, you may run into the following error:
+
+```bash
+Error: building AzureRM Client: obtain subscription() from Azure CLI: parsing json result from the Azure CLI: waiting for the Azure CLI: exit status 1: ERROR: No subscription found. Run 'az account set' to select a subscription.
+with provider["registry.terraform.io/hashicorp/azurerm"],
+on provider.tf line 14, in provider "azurerm":
+14: provider "azurerm" {
+
+```
+
+This is usually happen when terraform is unable to find Azure Subscription. The core issue behind this is that user is not logged to Azure account.
+
+In order to fix this you must login to Azure using the following command:
+
+```bash
+az login
+```
+
+## 4. Notes
+
+- The azurerm_storage_container resource block can not be used due to a bug in the terraform provider. Hence the deployment is done using ARM template for storage containers
+- The terraform script in this sample uses [startswith](https://developer.hashicorp.com/terraform/language/functions/startswith) function which needs minimum terraform version of 1.3.x
+- The sample script expects the resource group to be created before it is executed.
```

Comment:
Why don't we just create the resource group as well?

Reply:
Done